.github/workflows/build.yml

```diff
@@ -0,0 +1,162 @@
+              '{ "QMAKESPEC": "linux-g++",
+                 "EVAL": "export CC=gcc && export CXX=g++",
+                 "CFLAGS": "-Os",
+                 "LDFLAGS": "-Wl,--no-undefined -lm",
+                 "QMAKEFLAGS": "-config release",
+                 "QT_NO_CPU_FEATURE": "rdrnd"
+              }'
+          }
+        ]
+        include:
+          - os: macos-11
+            build_cfg: { "name": "clang",
+                         "flags":
+                           '{ "QMAKESPEC": "macx-clang",
+                              "CFLAGS": "-Oz",
+                              "QMAKEFLAGS": "-config debug",
+                              "MAKEFLAGS": "-s",
+                              "TESTARGS": "-silent",
+                              "PATH": "/usr/local/opt/qt5/bin:$PATH"
+                            }'
+                        }
+
+    # Default job name is too long to be visible in the "Checks" tab.
+    name: ${{ matrix.os }}/${{ matrix.build_cfg.name }}
+    # The type of runner that the job will run on
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Clone tinycbor
+      uses: actions/checkout@v4
+
+    - name: install Linux software
+      if: matrix.os == 'ubuntu-latest'
+      run: |
+        # Need a recent Valgrind, otherwise debug info cannot be read.
+        sudo snap install valgrind --classic
+        sudo apt-get update
+        sudo apt-get install -y --no-install-recommends \
+          doxygen \
+          jq \
+          libc6-dbg \
+          libcjson-dev \
+          libfuntools-dev \
+          qtbase5-dev
+
+    - name: install macOS software
+      if: matrix.os == 'macos-11'
+      run: |
+        # Doxygen 1.9.7 is broken with ifdefs again, install 1.9.4 which works.
+        wget https://raw.githubusercontent.com/Homebrew/homebrew-core/41828ee36b96e35b63b2a4c8cfc2df2c3728944a/Formula/doxygen.rb
+        brew install doxygen.rb
+        rm doxygen.rb
+        brew install qt5 cjson
+
+    # Valgrind takes a long time to build, so put in a separate step
+    # to make it easy to disable.
+    - name: install macOS valgrind
+      if: matrix.os == 'macos-11'
+      run: |
+        # 3 cores on CI runners, so use make -j6 for homebrew.
+        export HOMEBREW_MAKE_JOBS=6
+        brew tap LouisBrunner/valgrind
+        brew install --HEAD LouisBrunner/valgrind/valgrind
+
+    # Install in /usr/bin so the ccache action gets the expected environment.
+    - name: install ccache
+      if: matrix.os == 'ubuntu-latest'
+      run: |
+        wget -nv https://github.com/ccache/ccache/releases/download/v4.8.3/ccache-4.8.3-linux-x86_64.tar.xz
+        sudo tar xf ccache-4.8.3-linux-x86_64.tar.xz -C /usr/bin --strip-components=1 --no-same-owner ccache-4.8.3-linux-x86_64/ccache
+        rm -f ccache-*-linux-x86_64.tar.xz
+
+    - name: ccache
+      uses: hendrikmuhs/ccache-action@v1.2.10
+      with:
+        key: compilation-${{ matrix.os }}-${{ matrix.build_cfg.name }}
+        max-size: 160M
```

Comment:
TinyCBOR is tiny. How much do we gain by having this cached?

Reply:
Last time I looked, the Github CI machines were dual core Xeon with a low clock frequency and slow IO. So the gain is noticeable.

Reply:
Let me see if I am allowed to accept this third-party action. I think "github.com/intel" may have disabled all third-party actions.

Reply:
Sorry, I can't accept this action. It needs to be "actions/" only or a few options.

Reply:
Removed.